libnss/src/host.rs

```diff
@@ -20,10 +20,10 @@ pub enum Addresses {
     V6(Vec<Ipv6Addr>),
 }
 
-impl Host {
-    pub unsafe fn to_c_hostent(self, hostent: *mut CHost, buffer: &mut CBuffer) {
-        (*hostent).name = buffer.write_str(self.name);
-        (*hostent).h_aliases = buffer.write_strs(&self.aliases);
+impl ToC<CHost> for Host {
```

Comment:
The reason for this `ToC` trait being generic in this way is that it will allow us to also implement `impl ToC<GaihAddrTuple> for Host` to implement `gethostbyname_r4`, which uses a different C-struct for the newer API.